src/main/java/io/github/bobocodebreskul/context/registry/SimpleBeanDefinitionRegistry.java

```diff
@@ -33,32 +56,44 @@ public String[] getAliases(String name) {
 
   @Override
   public void registerBeanDefinition(String beanName, BeanDefinition beanDefinition) {
-
+    if (beanDefinitionMap.containsKey(beanName)) {
+      throw new BeanDefinitionDuplicateException(
+          "Cannot registered a beanDefinition with name %s because a beanDefinition with that name is already registered%n".formatted(
+              beanName));
+    } else {
+      beanDefinitionMap.put(beanName, beanDefinition);
+    }
   }
 
   @Override
   public void removeBeanDefinition(String beanName) throws NoSuchBeanDefinitionException {
-
+    beanDefinitionMap.remove(beanName);
   }
 
   @Override
   public BeanDefinition getBeanDefinition(String beanName) throws NoSuchBeanDefinitionException {
-    return null;
+    return beanDefinitionMap.get(beanName);
   }
 
   @Override
   public boolean containsBeanDefinition(String beanName) {
-    return false;
+    return beanDefinitionMap.containsKey(beanName);
   }
 
   @Override
   public String[] getBeanDefinitionNames() {
-    return new String[0];
+    Set<String> beanDefinitionNamesSet = beanDefinitionMap.keySet();
```

Comment:
You can make use of Collection interface method `toArray()` here. 
For example:
```return beanDefinitionMap.keySet().toArray(String[]::new);```